include/af/event.h

```diff
@@ -0,0 +1,62 @@
+/*******************************************************
+ * Copyright (c) 2014, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+
+#include <af/defines.h>
+
+#if AF_API_VERSION >= 37
+
+typedef void* af_event;
+
+#ifdef __cplusplus
+namespace af {
+
+/// A simple RAII wrapper for af_event
+class AFAPI event {
+    af_event e_;
+    bool preserve_ = false;  // Preserve the event after wrapper deletion
```

Comment:
I can zero out `e_`, but I actually added this because we can only return one `void*` from `MemoryManager::alloc`, meaning we have to put an `af_event` inside an `af_buffer_info` then return that `af_buffer_info`. Since both `af::event` and `af::buffer_info` will fall out of scope if we use them in a cpp implementation (like the internal one), we have to have a way to ensure their lifetimes continue after `alloc` returns else there's no point in doing any RAII/we can't use the abstraction.

Reply:
Sounds good; I'll remove `preserve_`, although we won't have an immediate need for that case anyways if I switch back to the C API/`af_event` and `af_buffer_info` in the default memory manager implementation.